### PR TITLE
Add test to contract

### LINKED ISSUE #18: Create test for the contract
ℹ️ This issue is part of Epic #16 

## Objective

Test all contract functions to provide more security at the time of deployment.

## Checklist

- [ ] Create tests

### PR DESCRIPTION
**Pull request by Odraxs**
## Description

With `v0.6` now we can test out contract without utils, also we changed some integer types because they produce some errors when running the test or invoking the contract from the futurenet 

Some of the tested functions in the `Futurenet`:

```
soroban contract invoke \
    --wasm target/wasm32-unknown-unknown/release/mtk_automation.wasm \
    --secret-key <SECRET-KEY> \
    --rpc-url https://horizon-futurenet.stellar.cash:443/soroban/rpc \
    --network-passphrase 'Test SDF Future Network ; October 2022' \
    --id <CONTRACT-ID> \
    --fn initialize \
    -- \
    --admin GDS7L6N7VK2LFTCZN4M3E2IAHWSV7VISZ3QFH3WB3L4ETCQANQWD3E2I \
    --org_name 'Kommit' \
    --offsets '{"thank": {"i32": 30} }' \
    --fund_amount 10000 \
    --token_c_id '<TOKEN-ID>'

soroban contract invoke \
    --wasm target/wasm32-unknown-unknown/release/mtk_automation.wasm \
    --secret-key <SECRET-KEY> \
    --rpc-url https://horizon-futurenet.stellar.cash:443/soroban/rpc \
    --network-passphrase 'Test SDF Future Network ; October 2022' \
    --id <CONTRACT-ID> \
    --fn reward_m \
    -- \
    --admin_address GDS7L6N7VK2LFTCZN4M3E2IAHWSV7VISZ3QFH3WB3L4ETCQANQWD3E2I \
    --to GDPZQCLCUNOAXVOPRKBJPZFAN3CQSAOPOIRIJ4ERAFFLRY4VOAQRLCQR \
    --o_type thank
```

Closes #18 and  #14 
Because the issue #14 solution was discussed and resolved here

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by Odraxs** [comment]
Probably we need to define how to call the function that gives the `compensation`/`rewards` so we can define how to rename  `reward_m` and `c_type`/`r_type` better

File: mtk-automation/src/lib.rs [hunk 1]
```diff
@@ -38,7 +38,7 @@ pub trait OrganizationContractTrait {
     fn org_name(env: Env) -> Symbol;
 
     /// fund contract balance for the organization
-    fn fund_c(env: Env, approval_sign: Address);
+    fn fund_c(env: Env, approval_address: Address);
```
Comment (L-Zuluaga) [inline mtk-automation/src/lib.rs:41]: What do you think about using `admin_address` instead?

File: mtk-automation/src/contract_actions/reward.rs [hunk 2]
```diff
@@ -6,31 +6,31 @@ use crate::contract_actions::member;
 use crate::contract_actions::token_operation;
 use soroban_sdk::{Address, Env, Map, Symbol};
 
-pub fn set_rewards(env: &Env, compensate_types: &Map<Symbol, u32>) {
+pub fn set_rewards(env: &Env, compensate_types: &Map<Symbol, i32>) {
```
Comment (L-Zuluaga) [inline mtk-automation/src/contract_actions/reward.rs:9]: I think we can follow @einerzg suggestion and use the word **offset** for naming this, so we can put everything in those terms like:
`set_offsets`, `offset_a_member`, `is_offset_valid`, etc

What do you think? 🤔

**Review by L-Zuluaga — CHANGES_REQUESTED** [review]

File: mtk-automation/src/contract_actions/offset.rs [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+use crate::contract_actions::datakey::DataKey;
+use crate::contract_actions::member;
+use crate::contract_actions::token_operation;
+use soroban_sdk::{Address, Env, Map, Symbol};
+
+pub fn set_offset(env: &Env, offset_types: &Map<Symbol, i32>) {
+    env.storage().set(&DataKey::Offsets, offset_types);
+}
+
+pub fn offset_member(env: &Env, admin_address: &Address, to: &Address, offset_type: &Symbol) {
```
Comment (L-Zuluaga) [inline mtk-automation/src/contract_actions/offset.rs:13]: Maybe `offset_a_member` describes better what the function does?

**Review by L-Zuluaga — APPROVED** [review]
LGTM! Excelent job 🚀 
I left a small comment just in case you want to take it into account :)

**Review by einerzg — CHANGES_REQUESTED** [review]
Checking the coverage I get that these two (`revoke_m` & `get_tc_id`) functions were not covered. Please check. 
Screen Shot 2023-02-21 at 10 02 43 PM

Ideal to have the coverage at 100% is now at 77.57%.

**Comment by Odraxs** [comment]
> Checking the coverage I get that these two (`revoke_m` & `get_tc_id`) functions were not covered. Please check. Screen Shot 2023-02-21 at 10 02 43 PM
> 
> Ideal to have the coverage at 100% is now at 77.57%.

@einerzg Sure we can test `get_tc_id`, but `revoke_m` is not possible right now because it has a `todo!` which works like a `panic!` that gives a reminder of a not fully implemented functionality

File: mtk-automation/src/test.rs [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+}
+
+fn create_and_init_token_contract(env: &Env, admin_id: &Address) -> (BytesN<32>, token::Client) {
+    let id = env.register_stellar_asset_contract(admin_id.clone());
+    let token = token::Client::new(env, &id);
+    (id, token)
+}
+
+#[test]
+fn happy_path() {
```
Comment (L-Zuluaga) [inline mtk-automation/src/test.rs:18]: What do you think about changing this name for something like: **succesfully_add_and_offset_a_member** or something that desribes better what we are testing in this specific case? 🤔

**Review by L-Zuluaga — DISMISSED** [review]
Everything LGTM, I left a small comment in case you want to take it into account

File: mtk-automation/src/test.rs (hunk 4 above)
Comment (Odraxs) [reply mtk-automation/src/test.rs:18]: @L-Zuluaga sounds good at least until the swap functionality is fully implemented and we will want to test revoke there, I'm changing it to the suggested name until that implementation

**Review by L-Zuluaga — APPROVED** [review]

**Review by einerzg — APPROVED** [review]

**Pull request merged**